spring-integration-core/src/main/java/org/springframework/integration/support/management/micrometer/MicrometerMetricsFactory.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2015-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.support.management.micrometer;
+
+import java.util.function.Function;
+
+import org.springframework.integration.support.management.AbstractMessageChannelMetrics;
+import org.springframework.integration.support.management.AbstractMessageHandlerMetrics;
+import org.springframework.integration.support.management.DefaultMessageChannelMetrics;
+import org.springframework.integration.support.management.DefaultMessageHandlerMetrics;
+import org.springframework.integration.support.management.MessageSourceMetrics;
+import org.springframework.integration.support.management.MessageSourceMetricsConfigurer;
+import org.springframework.integration.support.management.MetricsFactory;
+import org.springframework.util.Assert;
+
+import io.micrometer.core.instrument.MeterRegistry;
+
+/**
+ * Micrometer implementation.
```

Comment:
I think need more JavaDocs, e.g. explain the reason of the `MessageSourceMetricsConfigurer`.